### PR TITLE
RSpec Testing for Enumerable Methods

### PR DESCRIPTION
**Pull request by Georjane**
We created different examples to test all Enumerable methods using RSpec (Test Driven Development)
- my_each
- my_each_with_index
- my_select
- my_all?
- my_any?
- my_none?
- my_count
- my_map
- my_inject

### CODE REVIEW

**Comment by Georjane** [comment]
RSpec works just fine from the terminal but we don't know how to solve the RSpec setup error below

```
/opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/rubygems.rb:283:in `find_spec_for_exe': Could not find 'bundler' (2.1.4) required by your /home/runner/work/Enumerable-Methods/Enumerable-Methods/Gemfile.lock. (Gem::GemNotFoundException)
To update to the latest version installed on your system, run `bundle update --bundler`.
To install the missing version, run `gem install bundler:2.1.4`
	from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/rubygems.rb:302:in `activate_bin_path'
	from /opt/hostedtoolcache/Ruby/2.6.6/x64/bin/bundle:23:in `<main>'
```

File: spec/enum_spec.rb [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  let(:arr) { (1..5) }
+  let(:animals) { %w[ant bear cat] }
+  let(:array) { [1, 2, 4, 2] }
+  let(:inj_arr) { (5..10) }
+
+  describe '#my_each' do
+    it 'returns itself' do
+      expect(arr.my_each { |x| x * 2 }).to eql(1..5)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enum_spec.rb]: Let test for when:
- [x] returns an Enumerator if no block is given
- [x] calls the given block once for each element in self

File: spec/enum_spec.rb [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+  describe '#my_each_with_index' do
+    it 'returns itself' do
+      expect(arr.my_each_with_index { |x, y| x * y }).to eql(1..5)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enum_spec.rb]: Let test for when:
- [x] returns an Enumerator if no block is given
- [x] calls the given block once for each element in self

File: spec/enum_spec.rb [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    end
+
+    it 'Returns true if the block never returns false or nil.' do
+      expect([nil, true, 99].my_all?).to be false
+    end
+
+    it 'Returns true if the block never returns false or nil.' do
+      expect([].my_all?).to be true
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enum_spec.rb]: Let test for when:
- [x] when a Regex is passed as an argument

File: spec/enum_spec.rb [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    end
+
+    it 'Returns true if the block ever returns a value other than false or nil.' do
+      expect([nil, true, 99].my_any?).to be true
+    end
+
+    it 'Returns true if the block ever returns a value other than false or nil.' do
+      expect([].my_any?).to be false
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enum_spec.rb]: Let test for when:
- [x] when a Regex is passed as an argument

File: spec/enum_spec.rb [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    end
+
+    it 'Returns true if the block never returns true for all elements.' do
+      expect([nil, false].my_none?).to be true
+    end
+
+    it 'Returns true if the block never returns true for all elements.' do
+      expect([nil, false, true].my_none?).to be false
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enum_spec.rb]: Let test for when:
- [x] when a Regex is passed as an argument

File: spec/enum_spec.rb [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+  describe '#my_map' do
+    it 'Returns a new array with the results of running a block once for every element.' do
+      expect(arr.my_map { |i| i * i }).to eql [1, 4, 9, 16, 25]
+    end
+
+    it 'Returns a new array with the results of running a block once for every element.' do
+      expect(arr.my_map { 'cat' }).to eql %w[cat cat cat cat cat]
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enum_spec.rb]: Let test for when:
- [x] returns an Enumerator if no block is given

File: spec/enum_spec.rb [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+  describe '#my_inject' do
+    it 'Combines all elements by applying a binary operation, specified by a block or a symbol that names a method or operator.' do
+      expect(inj_arr.my_inject { |sum, n| sum + n }).to eql 45
+    end
+
+    it 'Combines all elements by applying a binary operation, specified by a block or a symbol that names a method or operator.' do
+      longest = animals.my_inject { |memo, word| memo.length > word.length ? memo : word }
+      expect(longest).to eql 'bear'
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enum_spec.rb]: Test for when :
- [x] when a symbol is specified with an initail value

**Review by kobiyoyo — COMMENTED** [review]
## Hi Team     
#### Great Job so far, but there are a few more things required. 
### General
- [x] Let's reopen this PR.
- [x]  Let's edit this pull request to Gitflow.
- [x] Check comments below 


### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi Team 👋

You've done quite well on this. 👍

However, let's fix this:

- [x] Kindly include in your readme instructions to run the tests. In addition, you have the steps in `Getting Started` in your readme mixed up, the user should first clone the project before `cd` into the project folder, not the other way round. Also, the folder to cd into is `Enumerable-Methods`, which is the actual name of your folder.

Kindly fix this issue and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: spec/enum_spec.rb [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+        expect(arr.my_map { 'cat' }).to eql %w[cat cat cat cat cat]
+      end
+    end
+
+    context 'When block given' do
+      it 'Returns to enumerator' do
+        expect(arr.my_map).to be_an Enumerator
+      end
+    end
+  end
```
Comment (mikenath223) [inline spec/enum_spec.rb]: - [x]  Also the wording of the context description here is wrong, it should read `When no block is given`. Please fix it.

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**


Don't forget to fill the project completion form👍
Good luck with the next project.

HAPPY CODING👨‍💻

**Pull request merged**